libs/cardano-ledger-test/bench/Bench/Cardano/Ledger/StakeDistr.hs

```diff
@@ -0,0 +1,388 @@
+                                            )
+                                        )
+                                    )
+                                    ( \registeredAggregated ->
+                                        bench "union+" $
+                                          let dState = (dpsDState . lsDPState . esLState . nesEs) nes
+                                           in whnf (rewards dState UM.∪+) registeredAggregated
+                                    )
+                                ]
+                            ],
+                          bench "mir rule" $ whnf (mirR globals) (nesEs nes),
+                          bench "epoch rule" $ whnf (epochR globals (nesEL nes + 1)) (nesEs nes),
+                          bench "calculatePoolDistr, improved (current) version" $ whnf calculatePoolDistr (getSnap nes),
+                          bgroup
+                            "calculatePoolDistr subparts"
+                            [ bench "poolStake" $
+                                whnf (calculatePoolStake (const True) (_delegations (getSnap nes))) (_stake (getSnap nes)),
+                              bench "old calculatePoolDistr" $
+                                whnf (oldCalculatePoolDistr (const True)) (getSnap nes)
+                            ]
+                        ]
+                    ]
+                ]
+          )
+    )
+
+getSnap :: NewEpochState (AlonzoEra StandardCrypto) -> SnapShot StandardCrypto
+getSnap nes = (_pstakeSet . esSnapshots . nesEs) nes
+
+getRewardUpdate :: NewEpochState era -> RewardUpdate (EraCrypto era)
+getRewardUpdate nes =
+  case (nesRu nes) of
+    SNothing -> error "No RewardUpdate in new epoch state"
+    SJust (Pulsing _ _) -> error "RewardUpdate in new epoch state is pulsing"
+    SJust (Complete ru) -> ru
+
+-- =====================================================
+
+-- ========================================================
+
+data RUArgs era = RUArgs !(EpochState era) !EpochNo !(RewardUpdate (EraCrypto era))
+
+encodeRUArgs :: RUArgs (AlonzoEra StandardCrypto) -> Encode ('Closed 'Dense) (RUArgs (AlonzoEra StandardCrypto))
+encodeRUArgs (RUArgs i b c) = Rec RUArgs !> To i !> To b !> To c
+
+decodeRUArgs :: Decode ('Closed 'Dense) (RUArgs (AlonzoEra StandardCrypto))
+decodeRUArgs = RecD RUArgs <! From <! From <! From
+
+instance ToCBOR (RUArgs (AlonzoEra StandardCrypto)) where toCBOR x = encode (encodeRUArgs x)
+
+instance FromCBOR (RUArgs (AlonzoEra StandardCrypto)) where fromCBOR = decode decodeRUArgs
+
+instance NFData (RUArgs (AlonzoEra StandardCrypto)) where
+  rnf (RUArgs a b c) = seq (rnf a) $ seq (rnf b) (rnf c)
+
+-- ==================================================================
+
+-- | The inital ShelleyGenesis structure, encoded in json format as a Haskell String
+shelleyGenesis :: ShelleyGenesis CurrentEra
+shelleyGenesis =
+  fromJust $
+    Aeson.decode $
+      lazyBytes
```

Comment:
This is unnecessary. Just enable `OverloadedStrings` function and it will work just fine. If in fact that string actually contained unicode characters, then this is as easy as `Text.encodeUtf8 . Text.pack`, where both functions come from `text` package.
```suggestion
```